### DISCUSSION TITLE
More Import templates

### DISCUSSION THREAD

**Discussion by mohan-the-octocat** [ananthakumaran/paisa#50] [General]
👋 there.

I've built out templates for importing content from some of the institutions that I work with. (IDFC First Bank & Axis Bank transaction statements)

Is there a way I can share them back here? 

```
### IDFC First Bank
{{#if (isDate ROW.A "DD/MM/YYYY")}}
  {{date ROW.A "DD/MM/YYYY"}} {{ROW.C}}
    {{predictAccount prefix="Expenses"}}:{{ROW.D}}	{{negate ROW.F}} INR
    Assets:Checking:IDFC-4925	{{trim ROW.F}} INR
{{/if}}
```

```
### Axis Bank
{{#if (isDate ROW.A "DD/MM/YY")}}
  {{date ROW.A "DD/MM/YY"}} {{ROW.B}}
  {{#if (or (eq (trim ROW.E) "0.00") (isBlank (trim ROW.E)))}}
    Expenses:{{trim ROW.C}}      {{trim ROW.D}} INR
    Assets:Checking:Axis-9183				  {{negate ROW.D}} INR
  {{else}}
    Assets:Checking:Axis-9183	{{trim ROW.E}} INR
    Income:{{trim ROW.C}}        -{{trim ROW.E}} INR
  {{/if}}
{{/if}}
```

Also, if it helps, I can volunteer time to build some of the docs around importers. Let me know!

**Comment by ananthakumaran** [comment]
If you share an anonymized sample CSV file here (or email me), I will add it to Paisa's built-in template. I will add the sample file to the [test suite](https://github.com/ananthakumaran/paisa/tree/master/fixture/import)

**Comment by mohan-the-octocat** [comment]
Both the transaction CSVs require some level of "data massaging" before they can be deemed usable.

There needs to be some documentation around how to get the files to be import ready... 

Wondering whats the best way we can capture and share that... Is there a planned workitem for generating import specific documentation?

**Reply by ananthakumaran** [reply]
> There needs to be some documentation around how to get the files to be import ready...

Hard to answer this without actually seeing any sample files. I have been able to get nearly all the files to work, even the CAS statement can be imported. If not possible, I can improve the import to handle it.

**Comment by dbza** [comment]
Adding some starting points for a few more templates. Please be warned I am only starting to understand the double entry ledger format and accounting in general. At least this works to read the data from the CSV as of now:


  ICICI credit card annual statement CSV:


```
{{#if (regexpTest ROW.A "^[0-9][0-9]-[A-Z][A-Z][A-Z]-[0-9][0-9]$"  )}}
 {{date ROW.A }} {{ROW.C}}
  {{#if (regexpTest ROW.G "^-")}}
  	{{#if (regexpTest ROW.C "Autodebit Payment Recd|INFINITY PAYMENT RECEIVED") }}
   		Assets:Checking {{ROW.F}} INR
    {{else}}
    	Income:Refund:CreditCard:ICICI                     {{ROW.F}} INR
    {{/if}}
  {{else}}
    {{predictAccount prefix="Expenses"}}		{{ROW.F}} INR
  {{/if}}
    Liabilities:CreditCard:ICICI
{{/if}}
```


ICICI SB account statement annual (for FY2022-23 and older)
```

{{#if (isDate ROW.C "DD/MM/YYYY")}}
  {{date ROW.C "DD/MM/YYYY"}} {{ROW.F}}
  {{#if (eq (trim ROW.H) "0")}}
    {{predictAccount prefix="Expenses"}}      {{trim ROW.G}} INR  
    Assets:Checking:ICICI
  {{else}}
    Assets:Checking:ICICI                      {{trim ROW.H}} INR
    {{predictAccount prefix="Income"}}
  {{/if}}
{{/if}}

```
